### ISSUE TITLE
ability to continue conversation or follow up

### ISSUE THREAD

**Issue by dominiksipowicz** [wong2/chatgpt-google-extension#10]
A link to ChatGPT can be used to continue the conversation

**Comment by wong2** [comment]
I'm not sure if this will continue the conversation or start a new one

**Comment by Plutone11011** [comment]
A standard link to the ChatGPT page would just start a new conversation. Maybe a way to approximate this would be to automate sending the same question through the input form, but even then ChatGPT doesn't always return the same answer.

**Comment by 0xdevalias** [comment]
Given ChatGPT has released conversation saving/continuing now, this might be more viable?

Previous chats are accessible using their UUID with a link format of `https://chat.openai.com/chat/:UUID`. So as long as that UUID is something that the extension has available to it at that time, should be fairly trivial to create a link?

That said, this will likely conflict with the recently released 'delete search chat' feature:

- https://github.com/wong2/chat-gpt-google-extension/issues/130
  - https://github.com/wong2/chat-gpt-google-extension/releases/tag/v1.9.0
    - https://github.com/wong2/chat-gpt-google-extension/commit/a65da0a5290a1150a4d24879daeab876f7f3cd0e

**Comment by wong2** [comment]
Yes, that's what I'm trying to implement yesterday at first, but met some 429 problems so I decided to go the deletion way temporarily. Will try this again later!

**Comment by GovindKrishnan** [comment]
+1 to this 😅 
![image](https://user-images.githubusercontent.com/74248475/210193409-b3cca70e-303b-4df9-bc6d-c5427643481a.png)

**Comment by Duskfen** [comment]
Dear Mr. @wong2 
I found a way how to handle follow-up questions. 
I would correctly implement it by myself but sadly i have little experience in typescript, browser-messaging and Data-Stream-handling. 

When becoming the first answer, ChatGPT sends a conversation_id, which you already use to delete conversations.
For follow-up questions, just include a: conversation_id in the message-body: 
```
        await fetchSSE('https://chat.openai.com/backend-api/conversation', { //shouldn't be here but should give you an idea how it could look
          method: 'POST',
          signal: controller.signal,
          headers: {
            'Content-Type': 'application/json',
            Authorization: `Bearer ${accessToken}`,
          },
          body: JSON.stringify({
            action: 'next',
            conversation_id: conversationId, //important change!
            messages: [
              {
                id: uuidv4(),
                role: "user",
                content: {
                  content_type: "text",
                  parts: ["can you elaborate on this?"] //TODO actual follow-up question
                }
              },
            ],
            model: 'text-davinci-002-render',
            parent_message_id: messageId,//take the uuidv4 from the previous generated
          }),
          onMessage(message: string) {
            console.debug('follow-up sse message:', message)
          },
        })
```

which results in something like this:
![grafik](https://user-images.githubusercontent.com/36416163/213266510-f1ad5252-2af3-4bbd-8ff1-aaccc8b30094.png)

![grafik](https://user-images.githubusercontent.com/36416163/213266648-a8700c9f-d93a-4410-a9be-28f45b17a0e2.png)

For better understanding, I uploaded you the changed background → index.tsx file.
[index.zip](https://github.com/wong2/chat-gpt-google-extension/files/10449578/index.zip)

hope I could help

**Comment by ElectraPort** [comment]
I know a bit of Typescript and have been wanting to contribute to this excellent project. Let me take a look and see if I can help.

**Comment by isarikaya** [comment]
Were you able to find any solution to continue the conversation?

**Comment by lxfater** [comment]
I think I solve this problem.
```javascript
    async talk(question: string, options: TalkOptions) {
        const { signal, cId, pId, onMessage } = options;
        const accessToken = await this.getToken();
        let text = '';
        let id = '';
        let conversationID = ''

        const basis = {
            action: "next",
            messages: [
              {
                id: uuidv4(),
                role: "user",
                content: {
                  content_type: "text",
                  parts: [question],
                },
              },
            ],
            model: "text-davinci-002-render-sha",
            parent_message_id: pId? pId : uuidv4(),
        }
        const body = cId ? {...basis, conversation_id: cId} : basis;
        return new Promise<{
            text: string;
            cId: string;
            pId: string;
        }>((resolve, reject) => {
          this.fetchSSE("https://chat.openai.com/backend-api/conversation", {
            method: "POST",
            signal,
            headers: {
              "Content-Type": "application/json",
              Authorization: `Bearer ${accessToken}`,
            },
            body: JSON.stringify(body),
            onMessage(message: string) {
              console.debug("sse message", message);
              if (message === "[DONE]") {
                resolve({
                    text,
                    cId:conversationID,
                    pId:id
                })
                return;
              } else {
                try {
                  const data = JSON.parse(message) as Rely
                  text = data.message?.content?.parts?.[0];
                  id = data.message.id
                  conversationID = data.conversation_id
                  if(onMessage) {
                    onMessage(text)
                  }
                } catch (error) {
                  console.error(error);
                }
              }
            },
          });
        })
    }
```
I will like to make some contribution.  Give me a chance. @wong2 @0xdevalias

**Comment by lxfater** [comment]
In case anyone wants to solve this problem. I create an npm package to bring the AI power of the ChatGPT to another webpage and it has the ability to continue the conversation.

https://github.com/lxfater/ai-bridge

**Comment by wong2** [comment]
Released in v2.0.0
